src/Juvix/Compiler/Core/Transformation/ComputeTypeInfo.hs

```diff
@@ -0,0 +1,93 @@
+module Juvix.Compiler.Core.Transformation.ComputeTypeInfo (computeTypeInfo) where
+
+import Data.HashMap.Strict qualified as HashMap
+import Juvix.Compiler.Core.Data.BinderList qualified as BL
+import Juvix.Compiler.Core.Extra
+import Juvix.Compiler.Core.Info.TypeInfo qualified as Info
+import Juvix.Compiler.Core.Transformation.Base
+
+-- | Computes the TypeInfo for each subnode.
+--
+-- Assumptions:
```

Comment:
Don't need to do anything here, but, what if, during the pipeline, we store in the infotables this information. If we see polymorphism, the turn on a flag, for example. Then, we don't assume, we check the conditions meet.